src/dynamic_batch_scheduler.cc

```diff
@@ -496,6 +508,16 @@ DynamicBatchScheduler::GetDynamicBatch()
       }
     }
 
+    if (CustomBatchEnabled()) {
+      bool should_include = false;
+      CustomBatchIncl(queue_.RequestAtCursor().get(), &should_include);
+      if (!should_include) {
+        curr_payload_->MarkSaturated();
+        send_now = true;
+        break;
+      }
+    }
```

Comment:
When will this get hit? It looks like lines 500-507 will not allow the user to take care of batch sizes of irregular inputs which is what I understand the ticket to be? I am probably missing something.

Reply:
Thanks for this feedback. The behavior of the underlying dynamic scheduler shouldn't be changed. So I assume those lines handle the ragged batching case already (looks like that function [does not always assert equality](https://github.com/triton-inference-server/core/blob/main/src/scheduler_utils.cc#L79-L80), additional comments [here](https://github.com/triton-inference-server/core/blob/main/src/scheduler_utils.h#L47-L49)). So this should already handle the ragged batching case, with this acting as an additional check to allow the user to add extra custom batching rules. Like in the original request to add a volume constraint.

I can take a look to confirm.

Reply:
That makes more sense. If the goal is to allow the user to define more constraints on the dynamic batcher rather then as a replacement, then this is correct. That's what I was missing.